python/example/single_cell_extracellular_potentials.py

```diff
@@ -122,100 +117,164 @@ def probes(self, gid):
 
 # run simulation for 500 ms of simulated activity and collect results.
 sim.run(tfinal=500)
 
-# extract time, V_m and I_m for each compartment
+# extract time, V_m, I_m and I_c for each CV
 V_m_samples, V_m_meta = sim.samples(v_handle)[0]
 I_m_samples, I_m_meta = sim.samples(i_handle)[0]
 I_c_samples, I_c_meta = sim.samples(c_handle)[0]
 
-
 # drop recorded V_m values and corresponding meta data of
 # zero-sized CVs (branch-point potentials)
 inds = np.array([m.dist != m.prox for m in V_m_meta])
 V_m_samples = V_m_samples[:, np.r_[True, inds]]
 V_m_meta = np.array(V_m_meta)[inds].tolist()
 
-
 # note: the cables comprising the metadata for each probe
 # should be the same, as well as the reported sample times.
 assert V_m_meta == I_m_meta
 assert (V_m_samples[:, 0] == I_m_samples[:, 0]).all()
 
-# prep recorded data for plotting
+# prep recorded data for plotting and computation of extracellular potentials
 time = V_m_samples[:, 0]
 V_m = V_m_samples[:, 1:].T
-I_m = I_m_samples[:, 1:].T
-I_c = I_c_samples[:, 1:].T
 
-
-# gather geometry of CVs and assign segments to each CV
-p = arbor.place_pwlin(morphology)
-x, y, z, d = [np.array([], dtype=float).reshape((0, 2))] * 4
-CV_ind = np.array([], dtype=int)  # tracks which CV owns segment
-for i, m in enumerate(I_m_meta):
-    segs = p.segments([m])
-    for j, seg in enumerate(segs):
-        x = np.row_stack([x, [seg.prox.x, seg.dist.x]])
-        y = np.row_stack([y, [seg.prox.y, seg.dist.y]])
-        z = np.row_stack([z, [seg.prox.z, seg.dist.z]])
-        d = np.row_stack([d, [seg.prox.radius * 2, seg.dist.radius * 2]])
-        CV_ind = np.r_[CV_ind, i]
+# Add stimulation current to transmembrane current to mimic sinusoid synapse
+# current embedded in the membrane.
+I_m = I_c_samples[:, 1:].T + I_m_samples[:, 1:].T  # (nA)
 
 
 ###############################################################################
-# compute extracellular potential using segment information
+# Compute extracellular potentials
 ###############################################################################
-cell_geometry = lfpykit.CellGeometry(
-    x=x,
-    y=y,
-    z=z,
-    d=d
-)
 
-# membrane voltages, transmemrbane current and corresponding times
-cell_geometry.V_m = V_m  # mV
-# nA, sum stimulation and transmembrane current to mimic sinusoid synapse
-cell_geometry.I_m = I_m + I_c
-cell_geometry.time = time  # ms
-
-# locations where extracellular potential is predicted
-dx = 1
-dz = 1
-axis = np.round([x.min() - 10, x.max() + 10, y.min() - 10, y.max() + 10])
-# axis = np.round(axis)
-X, Y = np.meshgrid(np.linspace(axis[0], axis[1], int(np.diff(axis[:2]) // dx) + 1),
-                   np.linspace(axis[2], axis[3], int(np.diff(axis[2:]) // dz) + 1))
-Z = np.zeros_like(X)
+# ## Compute extracellular potentials
+# First we define a couple of classes to interface the LFPykit
+# library (https://LFPykit.readthedocs.io, https://github.com/LFPy/LFPykit):
+class ArborCellGeometry(lfpykit.CellGeometry):
```

Comment:
Nice! `ArborCellGeometry` and `ArborLineSourcePotential` seem like snippets that might eventually live inside a utility grab bag in either project. Not that it is important to move them there now, but what are your thoughts on that?

Reply:
Indeed. I think these can be put in LFPykit in due time. Perhaps there is a Python subclassing trick for all other classes that are set up similar to `LFPykit.LineSourcePotential` such that each modified class do not have to be explicitly defined. The modified method `get_transformation_matrix` would be the same in each case, as would be the `__init__` routine.